### ISSUE TITLE
Unsuccesfull 'auto' upgrade of 0.24.2

### ISSUE THREAD

**Issue by Tamas-Toth-ebola** [hacs/integration#1084]
## Version of HACS

0.24.1 > 0.24.2

## Describe the bug
Sadly I do not have enough time for deep mind about background just would like to tell that this 0.24.2 upgrade was the only successful upgrade of HACS ever in my life.

I got a notice that there is an updated version of HACS (as I defined an automation for it at very long time ago). I just opened HA interface, went to HACS subpage, it told met that update available, I pressed on the **[Update]** or so button as always earlier the orange 'loader' line started to 'spinning' but instead of finish it some seconds later it seemed that went any successful process as after 5 minutes the upgrade did not finished yet. As seemed that the process not responded I restarted HA from the GUI. After restarting I got notice that HACS could not be started.

There is no problem, I will search for the background just wanted to report ASAP it as there never been such situation before.

If you have any tip what should I check for investigate the situation I could thanks for it in advance!

## Debug log

<details>
  <summary>Logs</summary>

```text
Paste your logs here.
```

</details>

**Comment by ludeeus** [comment]
Github is having issues.

**Comment by Tamas-Toth-ebola** [comment]
Thanks for your fast response!

Maybe in such case would be better any error or warning message if possible. But as I said no problem, I will correct everything I need if I had more time later.

**Comment by ludeeus** [comment]
There should be multiple entries in your log _file_

**Comment by Tamas-Toth-ebola** [comment]
Ok, so just to inform you:

I got only one HACS result in HA log:
`2020-04-23 15:46:33 ERROR (MainThread) [homeassistant.setup] Setup failed for hacs: Integration not found.`

So I checked if HACS available as 'custom component' in it's folder. There was no 'hacs' folder in 'custom components', so seemed under the upgrade the process could delete the old one but could not decompress (or something like process) the new one, and after I restarted HASS, HACS 'started' to missing.

Downloaded the latest from here, put into it's required place, restarted HASS and everything seem ok. The 'old' HACS configurations still available so nothing 'corrupted'.

So there is was not to big problem but could be better if HACS could verify if there is any problem before it's update process (of course if Github was responsible the result).

Thanks for your attention and of course for your efforts for this great HA 'plugin'!
